src/Faker/Provider/kk_KZ/Person.php

```diff
@@ -134,22 +177,68 @@ class Person extends \Faker\Provider\Person
     );
 
     /**
+     * @param  integer $year
+     *
+     * @return integer|null
+     */
+    private static function getCenturyByYear($year)
+    {
+        if ($year >= 2000 && $year <= DateTime::year()) {
+            return self::CENTURY_21ST;
+        } elseif ($year >= 1900) {
+            return self::CENTURY_20TH;
+        } elseif ($year >= 1800) {
+            return self::CENTURY_19TH;
+        }
+    }
+
+    /**
      * National Individual Identification Numbers
      *
      * @link   http://egov.kz/wps/portal/Content?contentPath=%2Fegovcontent%2Fcitizen_migration%2Fpassport_id_card%2Farticle%2Fiin_info&lang=en
+     * @link   https://ru.wikipedia.org/wiki/%D0%98%D0%BD%D0%B4%D0%B8%D0%B2%D0%B8%D0%B4%D1%83%D0%B0%D0%BB%D1%8C%D0%BD%D1%8B%D0%B9_%D0%B8%D0%B4%D0%B5%D0%BD%D1%82%D0%B8%D1%84%D0%B8%D0%BA%D0%B0%D1%86%D0%B8%D0%BE%D0%BD%D0%BD%D1%8B%D0%B9_%D0%BD%D0%BE%D0%BC%D0%B5%D1%80
+     *
      * @param  \DateTime $birthDate
+     * @param  integer   $gender
+     *
      * @return string 12 digits, like 780322300455
      */
-    public static function individualIdentificationNumber(\DateTime $birthDate = null)
+    public static function individualIdentificationNumber(\DateTime $birthDate = null, $gender = self::GENDER_MALE)
     {
         if (!$birthDate) {
-            $birthDate = \Faker\Provider\DateTime::dateTimeBetween();
+            $birthDate = DateTime::dateTimeBetween();
         }
 
-        $dateAsString       = $birthDate->format('ymd');
-        $genderAndCenturyId = (string) static::numberBetween(1, 6);
-        $randomDigits       = (string) static::numerify('#####');
+        $dateAsString     = $birthDate->format('ymd');
+        $totalPopulation  = rand(1000, 2000);
```

Comment:
never, ever use `rand` in Faker. See explanation in https://github.com/fzaninotto/Faker/blob/master/CONTRIBUTING.md